src/core/broadcaster.ts

```diff
@@ -58,7 +59,10 @@ export module Core {
      * @returns {Broadcaster} The calling Broadcaster
      */
     public broadcast(...args: any[]) {
-      this._key2callback.values().forEach((callback) => callback(this._listenable, args));
+      args.unshift(this._listenable);
```

Comment:
I'm nervous about this call...  We're directly affecting the array that's being passed in.